plugins/ui-update/components/UIDetails.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from "react";
+import { componentFromStream } from "@dcos/data-service";
+import { catchError, filter, map, startWith, switchMap } from "rxjs/operators";
+import { BehaviorSubject, combineLatest, merge, of } from "rxjs";
+
+import {
+  queryCosmosForUIVersions,
+  queryUIServiceForMetadata,
+  rollbackDCOSUIVersion,
+  updateDCOSUIVersion,
+  versionUpdateAvailable
+} from "#PLUGINS/ui-update/utils";
+
+import ConfigurationMapSection from "#SRC/js/components/ConfigurationMapSection";
+import { UIMetadata } from "#SRC/js/data/ui-update/UIMetadata";
+import { Package } from "#SRC/js/data/cosmos/Package";
+
+import AvailableUpdateRow from "#PLUGINS/ui-update/components/AvailableUpdateRow";
+import InstalledVersion from "#PLUGINS/ui-update/components/InstalledVersion";
+import InitialScreen from "#PLUGINS/ui-update/components/InitialScreen";
+import FallbackScreen from "#PLUGINS/ui-update/components/FallbackScreen";
+
+import {
+  EMPTY_ACTION,
+  UIAction,
+  UIActions
+} from "#PLUGINS/ui-update/types/UIAction";
+
+const uiServiceActions$ = new BehaviorSubject<UIAction>(EMPTY_ACTION);
+
+function onClickRollbackUI() {
+  rollbackDCOSUIVersion(uiServiceActions$);
+}
+
+function onClickStartUpdate(version: string) {
+  updateDCOSUIVersion(version, uiServiceActions$);
```

Comment:
🤔 It could, that would basically move the majority of the logic into this file instead of in the util functions. I had put it there to be able to unit test it and keep similar logic together. 

If it returned a stream then the util would basically just be wrapping the logic to make a mutation call. Which is what the query util functions do. I'm fine with that if its better to keep the utils unaware of the action stream.